### ISSUE TITLE
Docs: A single source of truth for process/policies/expectations/requirements

### ISSUE THREAD

**Issue by sed-i** [canonical/snapcraft#5042]
### What needs to get done

We should have a single source of truth for process/policies/expectations/requirements. Should include:
- Ideally snaps should be organic to the app. Try to first reach out to the upstream developers to have this snap published officially.
- If the snap is external to the repo and you're not the developer of the snapped app, namespace the snap like so: <user>-<app_name>
- At least 4 characters
- For Canonical employees, the snap should either be under a Canonical repository, or published through a personal store account.
- If you require a classically confined snap, it will likely be approved if it fits one of the following categories:
  - terminal emulators, multiplexers, and shells
  - etc.

### Why it needs to get done

Currently the information is segmented across systems and individuals, creating surprises in the snap registration process.

**Comment by dilyn-corner** [comment]
Snap provenance and names is covered [here](https://snapcraft.io/docs/registering-your-app-name) already, specifically...

> Do not prefix or suffix the name, for instance with your username or “-snap.”

> Ultimately, each name should be owned and published by members of the relevant project... However, if you are not associated with the project but want to help them create a snap, we welcome you to [join snapcrafters](https://forum.snapcraft.io/t/join-snapcrafters/1325), create the snap yourself, register the name and hand off to upstream projects when asked.

Caveat, of course:

> There is a single exception for having your username as a suffix in a snap name: an unofficial snap that has no chance of being handed over to the official project. This should be done with extreme caution...


There's also a documented process for classically confined snaps [here](https://snapcraft.io/docs/reviewing-classic-confinement-snaps).

**Comment by sed-i** [comment]
Thanks @dilyn-corner!

The snapcraft CLI seems to encourage prefixing:
![image](https://github.com/user-attachments/assets/9c663784-1322-45c4-8ac1-91ee4357bd73)

It would be nice if snapcraft cli and the webui point to the same reference. Currently, each has slightly different (and partial) messaging. More broadly, needing to contact team members for links to docs is suboptimal. We need to do something about discovery.

**Comment by dilyn-corner** [comment]
Ha. That's Very Funny:tm:.

Agreed on the discoverability side; Graham Morrison and I have been working on these problems for quite some time.

Perhaps @medubelko has some idea of a plan surrounding these documents?

**Comment by medubelko** [comment]
@dilyn-corner @sed-i I agree that the discoverability isn't good and we should consolidate or remove the redundancies here.

@degville Do you know whether these pages de-indexed deliberately?

**Comment by mr-cal** [comment]
The [registration doc](https://snapcraft.io/docs/registering-your-app-name) has been [migrated to RTD](https://github.com/canonical/snapcraft/pull/5360).  I think the outstanding work here is to:
* Provide a link from the `snapcraft register` command to the RTD how-to.
* Update the RTD doc to recommend prefixing your username.
  * Prefixing aligns with the [tutorial](https://github.com/canonical/snapcraft/pull/5422), the `snapcraft register` help text, and other *craft apps

**Comment by mr-cal** [comment]
Now the outstanding work is:

- Provide a link from the `snapcraft register` command to [how-to](https://documentation.ubuntu.com/snapcraft/stable/how-to/publishing/register-a-snap/).

I think something along the lines of:
```
Go to https://documentation.ubuntu.com/snapcraft/stable/how-to/publishing/register-a-snap for more information on registering a snap.
```

This should show up whether the user confirms or denies the prompt to register a name.

**Comment by medubelko** [comment]
This will be covered in https://github.com/canonical/open-documentation-academy/issues/303, if it's completed first.
